qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1393,6 +1393,39 @@ def substitute_node(self, node, op, inplace=False):
             self._decrement_op(node.op)
         return new_node
 
+    def separable_circuits(self) -> List["DAGCircuit"]:
+        """Separate the dag circuit into its weakly connected components.
+
+        The global phase information in `self` will not be maintained in the
+        subcircuits returned by this method.
+        """
+        connected_components = rx.weakly_connected_components(self._multi_graph)
+
+        # Collect each disconnected subgraph
+        disconnected_subgraphs = []
+        for components in connected_components:
+            disconnected_subgraphs.append(self._multi_graph.subgraph(list(components)))
+
+        # Helper function for ensuring rustworkx nodes are returned in lexicographical,
+        # topological order
+        def _key(x):
+            return x.sort_key
+
+        # Create new DAGCircuit objects from each of the rustworkx subgraph objects
+        decomposed_dags = []
+        for subgraph in disconnected_subgraphs:
+            new_dag = self.copy_empty_like()
```

Comment:
`copy_empty_like` will also copy `global_phase`, so we should explicitly set `new_dag.global_phase = 0` somewhere below.